analysers/trials/throughput.py

```diff
@@ -13,52 +13,31 @@
 from pyspark import SparkConf
 from pyspark import SparkFiles
 
-def computeThroughput(dataRDD):
-    if(dataRDD.isEmpty()):
-        return (None, None)
-    
-    data = dataRDD.map(lambda r: (r['start_time'], r['end_time'])) \
-            .collect()
-    
-    smallest = None
-    for d in data:
-        t = d[0]
-        if smallest == None:
-            smallest = t
-        elif t != None:
-            if t < smallest:
-                smallest = t
-            
-    largest = None
-    for d in data:
-        t = d[1]
-        if largest == None:
-            largest = t
-        elif t != None:
-            if t > largest:
-                largest = t
-    
-    if largest is None or smallest is None:
-        return None
-    
-    delta = largest - smallest
-    delta = delta.total_seconds()
+def createQuery(sc, cassandraKeyspace, experimentID, trialID):
+    queries = []
     
-    tp = len(data)/float(delta)
+    execTimes = sc.cassandraTable(cassandraKeyspace, "trial_execution_time")\
+            .select("process_definition_id", "execution_time") \
+            .where("trial_id=? AND experiment_id=?", trialID, experimentID) \
+            .cache()
+    numProcesses = sc.cassandraTable(cassandraKeyspace, "trial_number_of_process_instances")\
+            .select("process_definition_id", "number_of_process_instances") \
+            .where("trial_id=? AND experiment_id=?", trialID, experimentID) \
+            .cache()
     
-    return tp
-
-def createQuery(sc, dataRDD, experimentID, trialID):
-    queries = []
+    ex = execTimes.filter(lambda r: r["process_definition_id"] == "all").first()["execution_time"]
+    npr = numProcesses.filter(lambda r: r["process_definition_id"] == "all").first()["number_of_process_instances"]
     
-    tp = computeThroughput(dataRDD)
+    tp = float(npr)/float(ex)
     
     queries.append({"experiment_id":experimentID, "trial_id":trialID, "process_definition_id":"all", "throughput":tp})
     
-    processes = dataRDD.map(lambda a: a["process_definition_id"]).distinct().collect()
+    processes = execTimes.map(lambda a: a["process_definition_id"]).distinct().collect()
     
     for process in processes:
-        tp = computeThroughput(dataRDD.filter(lambda r: r['process_definition_id'] == process))
+        npr = numProcesses.filter(lambda r: r["process_definition_id"] == process).first()["number_of_process_instances"]
+        
+        tp = float(npr)/float(ex)
```

Comment:
Don't we have the execution time of each single process? We should maybe refer to that one here.